contrib/mesos/docs/ha.md

```diff
@@ -66,5 +86,417 @@ The command line parameters that affect the hash calculation are listed below.
 - `--proxy-*`
 - `--static-pods-config`
 
+## Cold-standby
+
+Setting up Kubernetes on Mesos in cold-standby mode is similar to Kubernetes in
+standalone mode described in [Kubernetes HA][1]. However, special attention is
+needed when setting up K8sm scheduler so that when the currently active
+scheduler crashes/dies, a new one can be instantiated and take over the work.
+More precisely, the new scheduler needs to be compatible with the executors
+that were started previously by the dead scheduler.
+
+### Environment Variables
+
+We will set up K8sm master on 2 nodes in HA mode. The same steps can be
+extended to set up more master nodes to deal with more concurrent failures. We
+will define a few environment variables first to describe the testbed
+environment.
+
+```
+MESOS_IP=192.168.0.1
+MESOS_PORT=5050
+
+ETCD_IP=192.168.0.2
+ETCD_PORT=4001
+
+K8S_1_IP=192.168.0.3
+K8S_2_IP=192.168.0.4
+K8S_APISERVER_PORT=8080
+K8S_SCHEDULER_PORT=10251
+
+NGINX_IP=192.168.0.5
+NGINX_APISERVER_PORT=80
+NGINX_SCHEDULER_PORT=81
+```
+
+Other than the 2 K8sm master nodes (`192.168.0.3` and `192.168.0.4`), we also
+define a Mesos master at `192.168.0.1`, an etcd server at `192.168.0.4`, and an
+Nginx server that load balances between the 2 K8sm master nodes. 
+
+### K8sm Container Image
+
+We are using podmaster to coordinate leadership selection amongst K8sm masters.
+However, podmaster needs to run in a container (preferably in a pod), and on
+the leader node, its podmaster will instantiate scheduler and controller
+manager also in their separate pods. The podmaster image is pre-built and can
+be obtained from `gcr.io/google_containers/podmaster`. An official image that
+containers the `km` binary to start apiserver, scheduler, and controller
+manager is not yet available. But it can be built fairly easily.
+
+```shell
+$ cat <<EOF >Dockerfile
+FROM ubuntu
+MAINTAINER Hai Huang <haih@us.ibm.com>
+RUN mkdir -p /opt/kubernetes
+COPY kubernetes/_output/dockerized/bin/linux/amd64/ /opt/kubernetes
+ENTRYPOINT ["/opt/kubernetes/km"]
+EOF
+$ cat <<EOF >build.sh
+#!/bin/bash
+K8SM_IMAGE_NAME=haih/k8sm
+git clone http://github.com/kubernetes/kubernetes
+cd kubernetes
+git checkout release-1.1
```

Comment:
the goal is to *eventually* keep everything in the world upstream. right
now we're maintaining our own 1.1 fork instead of pushing cherry-picks to
the upstream/release-1.1 branch. since you've referenced 1.1 here it should
point to the mesosphere/kubernetes repo (**and** the specific stable
release branch).

ideally, with the upstream release-1.2 stream we'll be much closer in sync
w/ the release cycle and cherry-picks will be easier to push .. so everyone
would be able to get the latest stable k8sm from the upstream/release-1.2
tree.

does that make sense?


On Wed, Feb 24, 2016 at 3:24 PM, Hai Huang <notifications@github.com> wrote:

> In contrib/mesos/docs/ha.md
> <https://github.com/kubernetes/kubernetes/pull/21894#discussion_r54000506>
> :
>
> > +manager is not yet available. But it can be built fairly easily.
> > +
> > +```shell
> > +$ cat <<EOF >Dockerfile
> > +FROM ubuntu
> > +MAINTAINER Hai Huang <haih@us.ibm.com>
> > +RUN mkdir -p /opt/kubernetes
> > +COPY kubernetes/_output/dockerized/bin/linux/amd64/ /opt/kubernetes
> > +ENTRYPOINT ["/opt/kubernetes/km"]
> > +EOF
> > +$ cat <<EOF >build.sh
> > +#!/bin/bash
> > +K8SM_IMAGE_NAME=haih/k8sm
> > +git clone http://github.com/kubernetes/kubernetes
> > +cd kubernetes
> > +git checkout release-1.1
>
> I can add a note here on both the repo location and the current stable
> branch. A related question is I remember seeing it somewhere that it was
> mentioned that mesosphere/kubernetes was not going to be maintained anymore
> other than just tracking issues, and all code changes will be pushed
> upstream to kubernetes/kubernetes. (mesosphere/kubernetes-mesos#356
> <https://github.com/mesosphere/kubernetes-mesos/issues/356>). Is this no
> longer the case?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/21894/files#r54000506>.
>